### PR TITLE
Introduce the `wcpay_terminal_payment_completed_order_status` filter

### PR DESCRIPTION
**Pull request by mordeth**
#### Changes proposed in this Pull Request
This pull request introduces the filter `wcpay_terminal_payment_completed_order_status`, which allows for the alteration of the order status from 'completed' to 'processing' following a successful terminal charge.

More context: pdfdoF-3XW-p2

#### Testing instructions
- Add a new plugin to your `mu-plugins` folder with the following code: `add_filter( 'wcpay_terminal_payment_completed_order_status', function() { return 'processing'; } )`.
- Add a product to your cart.
- Finish the order by paying with a card reader or a simulated In-Person payment order.
- Verify that the order status is `processing` instead of `completed`.

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7563` or branch name `add/ipp-order-status-filter` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/ipp-order-status-filter) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f30df27c2cca46bc739ee2ee018aece2c3b62e17
- Build time: 2023-11-07 17:28:26 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by elazzabi — APPROVED** [review]
Looks good 👍

File: includes/class-wc-payments-order-service.php [hunk 1]
```diff
@@ -347,7 +347,16 @@ public function mark_payment_dispute_closed( $order, $charge_id, $status ) {
...
 	public function mark_terminal_payment_completed( $order, $intent_id, $intent_status ) {
-		$this->update_order_status( $order, Order_Status::COMPLETED, $intent_id );
+		/**
+		 * Filters the order status value after a successful terminal payment.
+		 *
+		 * This filter can be used to override the order status from `completed` to `processing` after a successful terminal charge.
+		 *
+		 * @since 6.7.0
+		 */
+		$order_status = apply_filters( 'wcpay_terminal_payment_completed_status', Order_Status::COMPLETED );
```
Comment (anu-rock) [inline includes/class-wc-payments-order-service.php]: I'm not an expert at WP filter naming, but WDYT about dropping "status" - `wcpay_terminal_payment_completed`?

Payment completed sounds like an event. The word "status" seems to be redundant here.

Comment (anu-rock) [reply includes/class-wc-payments-order-service.php]: Another potentially naive comment:

In the absence of callbacks hooked to this event, will `$order_status` default to completed?

In your testing instructions, you've asked to add a filter like this:

```add_filter( 'wcpay_terminal_payment_completed_status', 'processing' )```

Looking at the [apply_filters](https://developer.wordpress.org/reference/functions/apply_filters/) documentation, isn't `processing` supposed to be a callback function?

Apologies for such beginner questions. I'm trying to review and learn :)

Comment (mordeth) [reply includes/class-wc-payments-order-service.php]: > I'm not an expert at WP filter naming, but WDYT about dropping "status" - wcpay_terminal_payment_completed?

As we modify the status itself, in my opinion, we should keep it. As suggested, I have changed the filter name to `wcpay_terminal_payment_completed_order_status`.

Comment (mordeth) [reply includes/class-wc-payments-order-service.php]: > Looking at the [apply_filters](https://developer.wordpress.org/reference/functions/apply_filters/) documentation, isn't processing supposed to be a callback function?

Good catch. I have updated the testing instructions with an anonymous function as a callback.

Comment (anu-rock) [reply includes/class-wc-payments-order-service.php]: Thanks for the changes. I'll check back when the PR is ready for review again.

**Review by anu-rock — APPROVED** [review]
LGTM after the latest changes. We can optionally add an automated test, but feel free to merge if urgent.

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ⚠️ Missing - would be a nice addition |
| Naming  | ✅ All good |
| Typing  | Not applicable |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Testing instructions  | ✅ All good |

**Pull request merged**